File: Chess/v2/Chess.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class Chess extends JFrame {
    public Chess() {
        add(new ChessPanel());
        setTitle("Visual Chess");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setResizable(false);
        pack();
        setLocationRelativeTo(null);
        setVisible(true);
    }

    public static void main(String[] args) {
        new Chess();
    }
}

class ChessPanel extends JPanel implements MouseListener {
    private final int TILE = 80;
    private String[][] board = new String[8][8];
    private int selectedRow = -1, selectedCol = -1;
    private boolean whiteTurn = true;
    private String message = "White to move";

    public ChessPanel() {
        setPreferredSize(new Dimension(640, 720));
        addMouseListener(this);
        setupBoard();
    }

    private void setupBoard() {
        board[0] = new String[]{"bR","bN","bB","bQ","bK","bB","bN","bR"};
        board[1] = new String[]{"bP","bP","bP","bP","bP","bP","bP","bP"};
        for (int r = 2; r < 6; r++) for (int c = 0; c < 8; c++) board[r][c] = "";
        board[6] = new String[]{"wP","wP","wP","wP","wP","wP","wP","wP"};
        board[7] = new String[]{"wR","wN","wB","wQ","wK","wB","wN","wR"};
    }

    protected void paintComponent(Graphics g) {
        Graphics2D g2 = (Graphics2D)g;
        g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);

        for (int r = 0; r < 8; r++) {
            for (int c = 0; c < 8; c++) {
                Color base = (r + c) % 2 == 0 ? new Color(238, 218, 185) : new Color(118, 82, 48);
                g2.setPaint(new GradientPaint(c*TILE, r*TILE, base.brighter(), c*TILE+TILE, r*TILE+TILE, base.darker()));
                g2.fillRect(c*TILE, r*TILE, TILE, TILE);

                if (r == selectedRow && c == selectedCol) {
                    g2.setColor(new Color(255, 230, 0, 160));
                    g2.fillRect(c*TILE, r*TILE, TILE, TILE);
                }

                drawPiece(g2, board[r][c], c*TILE, r*TILE);
            }
        }

        g2.setColor(new Color(20, 20, 20));
        g2.fillRect(0, 640, 640, 80);
        g2.setColor(Color.WHITE);
        g2.setFont(new Font("Serif", Font.BOLD, 24));
        g2.drawString(message, 20, 690);
    }

    private void drawPiece(Graphics2D g2, String piece, int x, int y) {
        if (piece.equals("")) return;

        boolean white = piece.charAt(0) == 'w';
        char type = piece.charAt(1);

        Color main = white ? new Color(245, 245, 235) : new Color(35, 35, 40);
        Color edge = white ? new Color(120, 120, 120) : new Color(230, 230, 230);

        g2.setColor(new Color(0, 0, 0, 80));
        g2.fillOval(x + 18, y + 58, 44, 12);

        g2.setColor(main);
        g2.fillOval(x + 20, y + 48, 40, 16);
        g2.fillRoundRect(x + 25, y + 28, 30, 30, 10, 10);

        if (type == 'P') {
            g2.fillOval(x + 27, y + 18, 26, 26);
        } else if (type == 'R') {
            g2.fillRect(x + 23, y + 18, 34, 16);
            g2.fillRect(x + 25, y + 12, 8, 10);
            g2.fillRect(x + 37, y + 12, 8, 10);
            g2.fillRect(x + 49, y + 12, 8, 10);
        } else if (type == 'N') {
            Polygon horse = new Polygon();
            horse.addPoint(x+25, y+52); horse.addPoint(x+31, y+20); horse.addPoint(x+43, y+12);
            horse.addPoint(x+56, y+25); horse.addPoint(x+50, y+31); horse.addPoint(x+58, y+48);
            horse.addPoint(x+45, y+54);
            g2.fillPolygon(horse);
            g2.setColor(edge);
            g2.fillOval(x+44, y+24, 4, 4);
        } else if (type == 'B') {
            g2.fillOval(x + 24, y + 12, 32, 35);
            g2.setColor(edge);
            g2.drawLine(x+40, y+15, x+34, y+38);
        } else if (type == 'Q') {
            g2.fillOval(x+24, y+20, 32, 28);
            for (int i = 0; i < 5; i++) {
                g2.setColor(main);
                g2.fillOval(x + 20 + i*9, y + 9 + (i%2)*3, 10, 10);
            }
        } else if (type == 'K') {
            g2.fillOval(x+24, y+20, 32, 28);
            g2.fillRect(x+37, y+8, 6, 20);
            g2.fillRect(x+30, y+14, 20, 5);
        }

        g2.setColor(edge);
        g2.setStroke(new BasicStroke(2));
        g2.drawOval(x + 20, y + 48, 40, 16);
        g2.drawRoundRect(x + 25, y + 28, 30, 30, 10, 10);
    }

    public void mousePressed(MouseEvent e) {
        int col = e.getX() / TILE;
        int row = e.getY() / TILE;
        if (row < 0 || row >= 8 || col < 0 || col >= 8) return;

        if (selectedRow == -1) {
            if (!board[row][col].equals("") && correctTurn(board[row][col])) {
                selectedRow = row; selectedCol = col;
                message = "Selected " + board[row][col];
            }
        } else {
            if (validMove(selectedRow, selectedCol, row, col)) {
                board[row][col] = board[selectedRow][selectedCol];
                board[selectedRow][selectedCol] = "";
                whiteTurn = !whiteTurn;
                message = whiteTurn ? "White to move" : "Black to move";
            } else message = "Illegal move";
            selectedRow = selectedCol = -1;
        }
        repaint();
    }

    private boolean correctTurn(String p) { return whiteTurn && p.charAt(0)=='w' || !whiteTurn && p.charAt(0)=='b'; }

    private boolean validMove(int sr, int sc, int er, int ec) {
        String p = board[sr][sc], t = board[er][ec];
        if (p.equals("") || (!t.equals("") && t.charAt(0) == p.charAt(0))) return false;
        int dr = er - sr, dc = ec - sc;
        char color = p.charAt(0), type = p.charAt(1);

        if (type == 'P') {
            int dir = color == 'w' ? -1 : 1, start = color == 'w' ? 6 : 1;
            if (dc == 0 && dr == dir && t.equals("")) return true;
            if (dc == 0 && sr == start && dr == 2*dir && t.equals("") && board[sr+dir][sc].equals("")) return true;
            if (Math.abs(dc) == 1 && dr == dir && !t.equals("")) return true;
        }
        if (type == 'R') return (dr == 0 || dc == 0) && clear(sr, sc, er, ec);
        if (type == 'B') return Math.abs(dr) == Math.abs(dc) && clear(sr, sc, er, ec);
        if (type == 'Q') return (dr == 0 || dc == 0 || Math.abs(dr) == Math.abs(dc)) && clear(sr, sc, er, ec);
        if (type == 'K') return Math.abs(dr) <= 1 && Math.abs(dc) <= 1;
        if (type == 'N') return Math.abs(dr) == 2 && Math.abs(dc) == 1 || Math.abs(dr) == 1 && Math.abs(dc) == 2;
        return false;
    }

    private boolean clear(int sr, int sc, int er, int ec) {
        int stepR = Integer.compare(er, sr), stepC = Integer.compare(ec, sc);
        int r = sr + stepR, c = sc + stepC;
        while (r != er || c != ec) {
            if (!board[r][c].equals("")) return false;
            r += stepR; c += stepC;
        }
        return true;
    }

    public void mouseReleased(MouseEvent e) {}
    public void mouseClicked(MouseEvent e) {}
    public void mouseEntered(MouseEvent e) {}
    public void mouseExited(MouseEvent e) {}
}

File: Chess/v2/Chess.py
import tkinter as tk

TILE = 80

class EnhancedChess:
    def __init__(self):
        self.root = tk.Tk()
        self.root.title("Enhanced Visual Python Chess")
        self.canvas = tk.Canvas(self.root, width=640, height=720)
        self.canvas.pack()

        self.board = [
            ["bR","bN","bB","bQ","bK","bB","bN","bR"],
            ["bP","bP","bP","bP","bP","bP","bP","bP"],
            ["","","","","","","",""],
            ["","","","","","","",""],
            ["","","","","","","",""],
            ["","","","","","","",""],
            ["wP","wP","wP","wP","wP","wP","wP","wP"],
            ["wR","wN","wB","wQ","wK","wB","wN","wR"]
        ]
        self.selected = None
        self.white_turn = True
        self.message = "White to move"

        self.canvas.bind("<Button-1>", self.click)
        self.draw()
        self.root.mainloop()

    def draw_piece(self, piece, x, y):
        if not piece:
            return
        white = piece[0] == "w"
        kind = piece[1]
        main = "#f5f5eb" if white else "#232328"
        edge = "#777777" if white else "#e6e6e6"

        self.canvas.create_oval(x+18, y+58, x+62, y+70, fill="#000000", outline="")
        self.canvas.create_oval(x+20, y+48, x+60, y+64, fill=main, outline=edge, width=2)
        self.canvas.create_rectangle(x+25, y+28, x+55, y+58, fill=main, outline=edge, width=2)

        if kind == "P":
            self.canvas.create_oval(x+27, y+18, x+53, y+44, fill=main, outline=edge, width=2)
        elif kind == "R":
            self.canvas.create_rectangle(x+23, y+18, x+57, y+34, fill=main, outline=edge, width=2)
            for sx in [25, 37, 49]:
                self.canvas.create_rectangle(x+sx, y+12, x+sx+8, y+22, fill=main, outline=edge)
        elif kind == "N":
            pts = [x+25,y+52,x+31,y+20,x+43,y+12,x+56,y+25,x+50,y+31,x+58,y+48,x+45,y+54]
            self.canvas.create_polygon(pts, fill=main, outline=edge, width=2)
            self.canvas.create_oval(x+44, y+24, x+48, y+28, fill=edge, outline="")
        elif kind == "B":
            self.canvas.create_oval(x+24, y+12, x+56, y+47, fill=main, outline=edge, width=2)
            self.canvas.create_line(x+40, y+15, x+34, y+38, fill=edge, width=2)
        elif kind == "Q":
            self.canvas.create_oval(x+24, y+20, x+56, y+48, fill=main, outline=edge, width=2)
            for i in range(5):
                self.canvas.create_oval(x+20+i*9, y+9+(i%2)*3, x+30+i*9, y+19+(i%2)*3, fill=main, outline=edge)
        elif kind == "K":
            self.canvas.create_oval(x+24, y+20, x+56, y+48, fill=main, outline=edge, width=2)
            self.canvas.create_rectangle(x+37, y+8, x+43, y+28, fill=main, outline=edge)
            self.canvas.create_rectangle(x+30, y+14, x+50, y+19, fill=main, outline=edge)

    def draw(self):
        self.canvas.delete("all")
        for r in range(8):
            for c in range(8):
                color = "#eedab9" if (r+c)%2 == 0 else "#765230"
                self.canvas.create_rectangle(c*TILE, r*TILE, c*TILE+TILE, r*TILE+TILE, fill=color, outline="")
                if self.selected == (r, c):
                    self.canvas.create_rectangle(c*TILE+3, r*TILE+3, c*TILE+TILE-3, r*TILE+TILE-3, outline="yellow", width=5)
                self.draw_piece(self.board[r][c], c*TILE, r*TILE)

        self.canvas.create_rectangle(0, 640, 640, 720, fill="#141414")
        self.canvas.create_text(120, 682, text=self.message, fill="white", font=("Georgia", 22, "bold"))

    def click(self, e):
        row, col = e.y // TILE, e.x // TILE
        if not (0 <= row < 8 and 0 <= col < 8):
            return

        if self.selected is None:
            if self.board[row][col] and self.correct_turn(self.board[row][col]):
                self.selected = (row, col)
                self.message = "Selected " + self.board[row][col]
        else:
            sr, sc = self.selected
            if self.valid_move(sr, sc, row, col):
                self.board[row][col] = self.board[sr][sc]
                self.board[sr][sc] = ""
                self.white_turn = not self.white_turn
                self.message = "White to move" if self.white_turn else "Black to move"
            else:
                self.message = "Illegal move"
            self.selected = None
        self.draw()

    def correct_turn(self, p):
        return (self.white_turn and p[0]=="w") or ((not self.white_turn) and p[0]=="b")

    def valid_move(self, sr, sc, er, ec):
        p = self.board[sr][sc]
        t = self.board[er][ec]
        if not p or (t and t[0] == p[0]):
            return False
        dr, dc = er-sr, ec-sc
        color, kind = p[0], p[1]
        if kind == "P":
            direction = -1 if color == "w" else 1
            start = 6 if color == "w" else 1
            if dc == 0 and dr == direction and not t: return True
            if dc == 0 and sr == start and dr == 2*direction and not t and not self.board[sr+direction][sc]: return True
            if abs(dc) == 1 and dr == direction and t: return True
        if kind == "R": return (dr == 0 or dc == 0) and self.clear(sr, sc, er, ec)
        if kind == "B": return abs(dr) == abs(dc) and self.clear(sr, sc, er, ec)
        if kind == "Q": return (dr == 0 or dc == 0 or abs(dr) == abs(dc)) and self.clear(sr, sc, er, ec)
        if kind == "K": return abs(dr) <= 1 and abs(dc) <= 1
        if kind == "N": return (abs(dr), abs(dc)) in [(2,1),(1,2)]
        return False

    def clear(self, sr, sc, er, ec):
        step_r = (er > sr) - (er < sr)
        step_c = (ec > sc) - (ec < sc)
        r, c = sr + step_r, sc + step_c
        while (r, c) != (er, ec):
            if self.board[r][c]:
                return False
            r += step_r
            c += step_c
        return True

EnhancedChess()
